src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -112,7 +112,7 @@ def validate_request_update_args(request_args, config, reqmgr_db_service, param)
             args_without_status['RequestPriority'] = int(args_without_status['RequestPriority'])
             if (lambda x: (x >= 0 and x < 1e6))(args_without_status['RequestPriority']) is False:
                 raise InvalidSpecParameterValue("RequestPriority must be an integer between 0 and 1e6")
-            if request_args["RequestStatus"] == "assignment-approved":
+            if "RequestStatus" in request_args:
```

Comment:
Actually, `RequestStatus` cannot possible be in this big `if` block. I suggest we remove L115-117